### ISSUE TITLE
eSpeak doesn't work with the Polish language

### ISSUE THREAD

**Issue by rbozan** [echogarden-project/echogarden#45]
I have a question, not sure where to post so I'll post it here.

I'm unsure how to align when the language is Polish. For example, using the following audio and transcript:

[Audio](https://minio.learnfeliz.com/game/audio/pol/137e72297035bdc60a194b163875ea49326fa32be00c2b5468e3fcaf2f06ddaf)
[transcript.txt](https://github.com/echogarden-project/echogarden/files/14632917/transcript.txt)

And the following command:

```bash
npx echogarden align ./audio.mp3  transcript.txt result.json --language=pl
```

The result is:
```
Echogarden v0.12.1

Prepare for alignment.. 45.4ms
Load alignment module.. 3.0ms
Create alignment reference with eSpeak.. Error: Word entry 0 already has phones before its start marker was seen. The words were: [ 'On', 'zawsze', 'robi', 'notatki', '.' ]
```

**Comment by rotemdan** [comment]
Thanks for the report.

Seems like something about the eSpeak output has an issue with this sentence, specific to the Polish language. It apparently breaks the order of the markers:

```
npx echogarden speak "On zawsze robi notatki." --engine=espeak --language=pl
```

If I try the same exact sentence with other languages like `en`, `de`, `fr`, `hu`, `sl`, etc. I'm not getting an error.

I will this investigate this further to see exactly what eSpeak produces for this sentence for `pl`.

**Comment by rotemdan** [comment]
eSpeak seems to be producing empty phoneme events for this input:

![Screenshot_1](https://github.com/echogarden-project/echogarden/assets/8589488/c734f1ef-22c2-4eff-a98f-52cdd755fc65)

This is odd. 

Possibly an eSpeak bug (there are many others). I'll change the code to ignore phoneme events with empty `id`.

**Comment by rotemdan** [comment]
Actually the empty phonemes are already ignored. 

The actual problem is that eSpeak only produced the first marker after the second word:

![Screenshot_2](https://github.com/echogarden-project/echogarden/assets/8589488/1db46c19-fdc4-4a59-be4b-a8ca8fa9b9a6)

The first two words are not part of the text, they are connected to special characters like ` ` and `|` that are supposed to be ignored.

It seems like eSpeak's Polish TTS decided to synthesize those special characters as words.

If you can read the IPA, maybe you can understand what it's saying in Polish?

**Comment by rbozan** [comment]
Not sure as I can't speak Polish myself.

**Comment by rotemdan** [comment]
That's okay, it doesn't matter.

The problem is not only with eSpeak, it's the way I'm using the vertical bar character (`|`) as a supposedly invisible separator to optionally ensure words are separated during synthesis. All languages I've tested ignored this character, apparently except Polish. 

I also used this character, in addition to `()` as a kind of a workaround for various bugs it has with markers.

I simply removed this extra `|` character in `0.12.2` (just published). I don't think it is really necessary anymore given I've already switched to a different approach.

Let me know if you find any other issues.
